### PR TITLE
feat(v1 Modal): added

### PR DESCRIPTION
**Pull request by ACoolmanTelicent**
Ticket: https://telicent.atlassian.net/browse/TELFE-435

### Goal

Add an opinionated Modal that exposes enough of its guts - that no one needs to stick their hands in

### Work Completed

https://github.com/user-attachments/assets/a64168de-f66d-4814-b214-925cfd62a949

- Create modal with a bunch of props removed
- Demo usage in storybook for data catalog

### Testing Method

Screenshot 2024-09-18 at 15 15 55

### Code Review Tips 

- I left more comments than normal,I wanna make my decision process around design-system very clear (and open to criticism/feedback)

### Engineering Notes

- The schema for the new Modal props is not good enough to use for schema-type-to-real-type comparisons

### Out-of-scope 

Pixel Perfection; In a bit of a rush.

---

<details>
  <summary><strong>🕵️ Review Guidance</strong></summary>

---

General guidance

- Generally, approve a PR if it makes the system better, even if it's not perfect. — [Google: The Standard of Code Review](https://google.github.io/eng-practices/review/reviewer/standard.html)
- The aim of both PR AUTHOR and PR REVIEWER is to get the code merged
- Aim for consensus, defined as _everyone can live with the outcome_

For PR REVIEWER:

1. Read the ticket & description
2. [Review the code](https://google.github.io/eng-practices/review/reviewer/looking-for.html)
3. Request any changes that are essential.
4. For non-essential comments:
   - Use prefixes, e.g. "**nit:** change to Pascal case"
     - **nit:** small, non-essential change
     - **obs:** just an observation, doesn't affect the PR
     - **idea:** a suggestion to think about
     - **q:** questions
   - Use modifiers, e.g. "**obs**`[pr-owner-resolve]`: Jim is also editing this file"
     - `pr-author-resolve` PR author can resolve after reading
     - `pr-author-delete` (rare) delete after reading to avoid confusion
5. try to add _at least_ a helpful comment per ~500 lines; less if the PR is already busy

For PR AUTHOR:

1. Aim for enough detail in PR description for things to go smoothly
2. After requested changes, [re-request a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#re-requesting-a-review) (so the PR shows up in [reviews-requested:@me](https://github.com/pulls?q=is%3Apr+is%3Aopen+archived%3Afalse+sort%3Aupdated-desc+review-requested%3A%40me+))

</details>

### CODE REVIEW

**Review by AlecsMissangu — APPROVED** [review]

File: src/v1/components/utils/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            display: "flex",
+            justifyContent: "flex-end",
+          }}
+        >
+          <Button
+            onClick={(event) => props.onClose?.(event, "closeButtonClick")}
+            sx={{
+              fontSize: 18,
+              // TODO ask how to color
+              // HOW use theme.palette.primary.main
```
Comment (ACoolmanTelicent) [inline src/v1/components/utils/Modal/Modal.tsx:72]: @AlecsMissangu Off-the-top-of-the-dom

Do you have an opinion on how should I colour this text?

File: src/v1/components/utils/Modal/Modal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        >
+          <div />
+        </body>
+      `);
+    });
+  });
+
+  test('calls onClose with "closeButtonClick" when Close button is clicked', () => {
+    render(<Modal {...props} />);
+    fireEvent.click(screen.getByRole("button", { name: /Close/i }));
```
Comment (davec504) [inline src/v1/components/utils/Modal/Modal.test.tsx:108]: Suggestion: Steer clear from regex where possible
```suggestion
    fireEvent.click(screen.getByRole("button", { name: 'Close' }));
```

Comment (davec504) [reply src/v1/components/utils/Modal/Modal.test.tsx:108]: Also aren't we supposed to use userEvent instead of fireEvent?

**Review by davec504 — APPROVED** [review]

File: src/v1/components/utils/Modal/Modal.test.tsx (hunk 2 above)
Comment (ACoolmanTelicent) [reply src/v1/components/utils/Modal/Modal.test.tsx:108]: Yep, Copy pasta

**Review by AlecsMissangu — APPROVED** [review]

**Pull request merged**